src/librustdoc/html/format.rs

```diff
@@ -527,21 +521,15 @@ fn tybounds(param_names: &Option<Vec<clean::GenericBound>>) -> impl fmt::Display
     })
 }
 
-impl<'a> HRef<'a> {
-    pub fn new(did: DefId, text: &'a str) -> HRef<'a> {
-        HRef { did: did, text: text }
-    }
-}
-
-impl<'a> fmt::Display for HRef<'a> {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if let Some((url, short_ty, fqp)) = href(self.did) {
+pub fn anchor(did: DefId, text: &str) -> impl fmt::Display + '_ {
+    WithFormatter::new(move |f| {
```

Comment:
I think `WithFormatter` is introduced several commits *ahead*, which is mildly confusing